### PR TITLE
Added caching for the max time and memory used in a submission.

### PR DESCRIPTION
**Pull request by deyordanov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1558

### CODE REVIEW

File: Services/UI/OJS.Services.Ui.Models/Submissions/FullDetailsPublicSubmissionsServiceModel.cs [hunk 1]
```diff
@@ -65,14 +63,41 @@ public void RegisterMappings(IProfileExpression configuration)
...
-            .ForMember(
-                d => d.MaxMemoryUsed,
-                opt => opt.MapFrom(s => s.TestRuns.Any() ? s.TestRuns.Max(testRun => testRun.MemoryUsed) : (long?)null))
-            .ForMember(
-                d => d.MaxTimeUsed,
-                opt => opt.MapFrom(s =>
-                    s.TestRuns.Any() ? s.TestRuns.Max(testRun => testRun.TimeUsed) : (int?)null))
+            .ForMember(x => x.MaxMemoryUsed, opt => opt.MapFrom(
+                s => GetMaxMemoryAndTimeUsed(s.TestRunsCache).MaxMemoryUsed))
+            .ForMember(x => x.MaxTimeUsed, opt => opt.MapFrom(
```
Comment (gogo4ds) [inline Services/UI/OJS.Services.Ui.Models/Submissions/FullDetailsPublicSubmissionsServiceModel.cs:68]: Is the parsing translate to SQL and done in the query? I suggest using `.AfterMap()`, as you only need the TestRunsCache, which is already present in the destination.

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/UI/OJS.Services.Ui.Models/Submissions/FullDetailsPublicSubmissionsServiceModel.cs (hunk 1 above)
Comment (deyordanov) [reply Services/UI/OJS.Services.Ui.Models/Submissions/FullDetailsPublicSubmissionsServiceModel.cs]: Done.

File: Services/UI/OJS.Services.Ui.Models/Submissions/FullDetailsPublicSubmissionsServiceModel.cs [hunk 2]
```diff
@@ -63,13 +63,20 @@ public void RegisterMappings(IProfileExpression configuration)
...
+            .ForMember(
+                x => x.MaxTimeUsed,
+                opt => opt.Ignore())
+            .ForMember(
+                x => x.MaxMemoryUsed,
+                opt => opt.Ignore())
+            .AfterMap((src, dest) =>
+            {
+                dest.MaxTimeUsed = GetMaxMemoryAndTimeUsed(src.TestRunsCache).MaxTimeUsed;
+                dest.MaxMemoryUsed = GetMaxMemoryAndTimeUsed(src.TestRunsCache).MaxMemoryUsed;
```
Comment (gogo4ds) [inline Services/UI/OJS.Services.Ui.Models/Submissions/FullDetailsPublicSubmissionsServiceModel.cs:78]: Why call the method twice?

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/UI/OJS.Services.Ui.Models/Submissions/FullDetailsPublicSubmissionsServiceModel.cs (hunk 2 above)
Comment (deyordanov) [reply Services/UI/OJS.Services.Ui.Models/Submissions/FullDetailsPublicSubmissionsServiceModel.cs]: Fixed.

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**